File: 1094-car-pooling/1094-car-pooling.cpp
class Solution {
public:
    bool carPooling(vector<vector<int>>& trips, int capacity) {
        vector<int> res(1001, 0);
        for(auto t:trips){
            int p = t[0], s=t[1], e=t[2];
            res[s]+=p;
            res[e]-=p;
        }
        
        int ans=res[0];
        for(int i=1;i<res.size();i++){
            res[i] += res[i-1];
            ans = max(ans, res[i]);
        }
        
        return ans <= capacity;
    }
};

File: 1094-car-pooling/1094-car-pooling.ts
function carPooling(trips: number[][], capacity: number): boolean {
    let res = new Array<number>(1001);
    for(let i=0;i<=1001;i++) res[i]=0;
    
    for(let t of trips){
        let p:number=t[0];
        let s:number = t[1];
        let e:number = t[2];
        
        res[s]+=p;
        res[e]-=p;
    }
    
    let maxVal = res[0];
    for(let i=1;i<=1001;i++){
        res[i] += res[i-1];
        maxVal = Math.max(maxVal, res[i]);
        if(maxVal>capacity) return false;
    }
    return true;
    
};